src/test/scala/mesosphere/marathon/api/validation/RunSpecValidatorTest.scala

```diff
@@ -92,33 +91,48 @@ class RunSpecValidatorTest extends UnitTest with ValidationTestLike {
       testValidId("/trailing/")
     }
 
-    "single dots in id '/test/.' pass schema and validation" in {
-      testInvalid("/test/.")
+    "single dots in id '/test/.' fails in validation" in {
+      val app = App(
+        id = "/test/.",
+        cmd = Some("true"))
+
+      validApp(app) should haveViolations("/id" -> "must not end with any of the following reserved keywords: restart, tasks, versions, ., ..")
```

Comment:
@zen-dog @timcharper I've adjusted these tests here, could you have another look at them? I'm not completley sure, especially about the ones with "/../someid" and "/./not.point.less" IDs...

Reply:
I'm of the opinion that the AbsolutePathId constructor should always remove the relative path elements

Reply:
I'm pretty sure it does, this test is about the validation of pathes in the raml.App

I don't think it would be a good idea to allow users to send `/test/.` as a pathId, and end up with `/test`

Reply:
Except that this would be a breaking API change